### PR TITLE
Mypybot/sync typeshed

### PR DESCRIPTION
**Pull request by koogoro**
Sync typeshed before 1.11 release.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
There's some new errors in mypy itself, I can look into fixing those over the next few days.

**Comment by koogoro** [comment]
Should I even merge this if it's going to cause errors in mypy checking mypy?

**Comment by JelleZijlstra** [comment]
No, the type checking issues need to be addressed first. I'll take a look now.

File: mypy/fastparse.py [hunk 1]
```diff
@@ -2041,8 +2041,10 @@ def visit_Subscript(self, n: ast3.Subscript) -> Type:
                 sliceval.col_offset = sliceval.lower.col_offset
         else:
             assert isinstance(n.slice, ast3.ExtSlice)
-            dims = copy.deepcopy(n.slice.dims)
+            dims = cast(list[ast3.expr], copy.deepcopy(n.slice.dims))
```
Comment (AlexWaygood) [inline mypy/fastparse.py:2044]: Py38 compat:

```suggestion
            dims = cast(List[ast3.expr], copy.deepcopy(n.slice.dims))
```

**Pull request merged**